src/components/query-viewer/query-viewer.tsx

```diff
@@ -0,0 +1,382 @@
+        !this.propsNotToPost.includes(key)
+      ) {
+        postBody +=
+          '<input type="hidden" name="' +
+          key +
+          '" value="' +
+          Object(this)[key] +
+          '" />';
+      }
+    }
+    postBody +=
+      '<input type="hidden" name="Width" value="' + this.getWidth() + '" />';
+    postBody +=
+      '<input type="hidden" name="Height" value="' + this.getHeight() + '" />';
+    postBody +=
+      '<input type="hidden" name="Elements" value="' +
+      this.getElements() +
+      '" />';
+    postBody +=
+      '<input type="hidden" name="Parameters" value="' +
+      this.getParameters() +
+      '" />';
+    return postBody;
+  }
+
+  private getParameters(): string {
+    let parametersValue = "";
+    const parameters = Array.from(
+      document.getElementsByTagName("gx-query-viewer-parameter")
+    );
+    parameters.forEach(parameter => {
+      parametersValue += parametersValue != "" ? "," : "";
+      parametersValue +=
+        "{Value:" +
+        encodeURIComponent(parameter.Value) +
+        ",Name:" +
+        parameter.Name +
+        "}";
+    });
+    parametersValue = "[" + parametersValue + "]";
+
+    return parametersValue;
+  }
+
+  private getElements(): string {
+    let elementsValue = "";
+    const elements = Array.from(
+      document.getElementsByTagName("gx-query-viewer-element")
+    );
+    elements.forEach(ax => {
+      let elementValue = "{";
+      elementValue += ' "Name": "' + ax.name + '"';
+      elementValue += ' ,"Title": "' + ax.elementTitle + '"';
+      elementValue += ' ,"Visible": "' + ax.visible + '"';
+      elementValue += ' ,"Type": ' + ax.type + '"';
+      elementValue += ' ,"Axis": "' + ax.axis + '"';
+      elementValue += ' ,"Aggregation": "' + ax.aggregation + '"';
+      elementValue += ' ,"DataField": ' + ax.dataField + '"';
+      if (ax.axisOrderType) {
+        elementValue += ', "AxisOrder": { "Type": "' + ax.axisOrderType + '"';
+        elementValue +=
+          ax.axisOrderValues != null
+            ? ', "Values": [' + ax.axisOrderValues + "]"
+            : "";
+        elementValue += "}";
+      }
+      if (ax.filterType) {
+        elementValue += ', "Filter": { "Type": ' + ax.filterType + '"';
+        elementValue +=
+          ax.filterValues != null
+            ? ', "Values": [' + ax.filterValues + "]"
+            : "";
+        elementValue += "}";
+      }
+      if (ax.expandCollapseType) {
+        elementValue +=
+          ', "ExpandCollapse": { "Type": "' + ax.expandCollapseType + '"';
+        elementValue +=
+          ax.expandCollapseValues != null
+            ? ' "Values": [' + ax.expandCollapseValues + "]"
+            : "";
+
+        elementValue += "}";
+      }
+      //TODO add grouping values
+      const formats = Array.from(
+        ax.getElementsByTagName("gx-query-viewer-element-format")
+      );
+
+      formats.forEach(format => {
+        elementValue += ' ,"Format":  {';
```

Comment:
Why build a JSON using String concatenation?
Maybe better to just use a json obj?  Less verbose and error prone. 

{
"Name": ax.name, 
"Title": ax.elementTitle,
...
}